src/KDSoapClient/KDSoapClientInterface.cpp

```diff
@@ -142,21 +144,37 @@ QNetworkRequest KDSoapClientInterfacePrivate::prepareRequest(const QString &meth
     return request;
 }
 
-QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const KDSoapHeaders &headers)
+QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const QString &soapAction, const KDSoapHeaders &headers)
 {
     KDSoapMessageWriter msgWriter;
     msgWriter.setMessageNamespace(m_messageNamespace);
     msgWriter.setVersion(m_version);
-    const QByteArray data = msgWriter.messageToXml(message, (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(), headers, m_persistentHeaders, m_authentication);
+
     QBuffer *buffer = new QBuffer;
-    buffer->setData(data);
+    if (m_sendSOAPActionInWSAAction) {
+        KDSoapMessage messageCopy = message;
+        KDSoapMessageAddressingProperties prop = message.messageAddressingProperties();
+        if (!prop.action().isEmpty())
+            qWarning() << "Overwriting the action addressing parameter with the SOAP action";
```

Comment:
I suggest printing both values, so the developer doing this gets an idea of where the problem comes from.

Reply:
Indeed, and I will also supress the warning if no overwrite is done (when the SOAP action and the action is identical).